core/src/storage/collections/btree_map/impls.rs

```diff
@@ -0,0 +1,1132 @@
+// Copyright 2018-2019 Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::storage::{
+    self,
+    alloc::{
+        Allocate,
+        AllocateUsing,
+        Initialize,
+    },
+    chunk::SyncChunk,
+    Flush,
+};
+#[cfg(feature = "ink-generate-abi")]
+use ink_abi::{
+    HasLayout,
+    LayoutField,
+    LayoutStruct,
+    StorageLayout,
+};
+use scale::{
+    Decode,
+    Encode,
+};
+#[cfg(feature = "ink-generate-abi")]
+use type_metadata::Metadata;
+use core::{
+    cmp::Ord,
+    ptr,
+};
+use super::search::{self, SearchResult};
+
+/// Each node in the tree has 2 * B children.
+pub(crate) const B: usize = 6;
```

Comment:
For the moment I just took the same parameters as [the Rust stdlib implementation](https://github.com/rust-lang/rust/blob/master/src/liballoc/collections/btree/node.rs#L42-L44). So yeah kinda arbitrarily, but I don't have a better idea until we can do benchmarks.

One thing that I just thought about: If we change these consts at any point this will require to have all existing `BTreeMap`s which are stored on-chain to be migrated. An alternative would be to store these params within the `BTreeMapHeader`. Hmm, but come to think of it this brings a lot of disadvantages, like us having to test and support all kinds of parameters and be endlessly downwards-compatible. Yeah, so from my POV we should stay with consts.